src/m_io.c

```diff
@@ -275,3 +275,40 @@ void M_MakeDirectory(const char *path)
     mkdir(path, 0755);
 #endif
 }
+
+#ifdef _WIN32
+#include "SDL.h"
+#endif
+
+char *M_getenv(const char *name)
+{
+#ifdef _WIN32
+    wchar_t *wenv = NULL, *wname = NULL;
+    char *env = NULL;
+    int size;
+
+    wname = ConvertUtf8ToWide(name);
+
+    if (!wname)
+    {
+        return NULL;
+    }
+
+    wenv = _wgetenv(wname);
+
+    free(wname);
+
+    if (!wenv)
+    {
+        return NULL;
+    }
+
+    size = (wcslen(wenv) + 1) * sizeof(wchar_t);
+
+    env = SDL_iconv_string("UTF-8", "UTF-16LE", (const char *)wenv, size);
```

Comment:
Then `env` must be freed somewhere afterwards.

Reply:
A string from `getenv` cannot be freed, so should we free `env`? There used to be this code:
```c
ret = strdup(env);
SDL_free(env);
return ret;
```
But I removed it.

Reply:
This, or we need a new function to be called after using `M_getenv()` which takes the returned string as an argument and either calls `free()` on it or does nothing. 🤷

Reply:
So we need to `strdup(env)`? But why? I think `SDL_iconv_string` copied it for us already.

Reply:
We are discussing the `ifndef _WIN32` case here. 😉 Currently, we have a function which returns an allocated `char *` on one arch and `const char *` on the others.

Reply:
Technically `getenv` returns `char *` without const.
OK, the cleaner way is to remove `M_getenv` and explicitly convert the path in places where `getenv` is called for the _WIN32 case.

Reply:
> Technically `getenv` returns `char *` without const.

Yes, right. Nevertheless the returned string is immutable and thus cannot get freed.

> OK, the cleaner way is to remove `M_getenv` and explicitly convert the path in places where `getenv` is called for the _WIN32 case.

Hm, not sure if this is cleaner. I was thinking about a function that is called whenever the string called by `M_getenv()` isn't used anymore.

```c
void M_FreeEnv(void *p)
{
#ifdef _WIN32
  if (p)
    free(p);
#endif
}
```

Reply:
Just that it's supposed to be `SDL_free()`, because that's how the string returned by `SDL_iconv_string()`is meant to be freed. And thus, in the non-Win32 case, we'd need to return `SDL_strdup(getenv())`.

Reply:
How about this solution: https://github.com/fabiangreffrath/woof/pull/667/commits/fda90168967b9c1605dea6b34c364215796a58e1

Reply:
Wow, very spohisticated, I like it! Could we also cache the case that `wenv == NULL` with an `env_vars[]` entry?